tests/rustdoc-ui/intra-doc/auxiliary/in-proc-item-comment.rs

```diff
@@ -0,0 +1,14 @@
+//@ force-host
+//@ no-prefer-dynamic
+#![crate_type = "proc-macro"]
+
+extern crate proc_macro;
+use proc_macro::TokenStream;
+
+mod view {}
+
+/// [`view`]
```

Comment:
This can never end up getting linked anyway as proc macros can't export anything other than `#[proc_macro]` and such. The `doc_link_resolutions` and `doc_link_traits_in_scope` handling code already skips encoding things for proc-macros in certain cases. Currently one of the requirements for this skipping is for the item to be non-local. I think removing the requirement that it is non-local would fix the ICE too and be a much better solution. You can't link to private items anyway and the only non-private items proc macros allow are `#[proc_macro]` and such.

https://github.com/rust-lang/rust/blob/717f5df2c308dfb4b7b8e6c002c11fe8269c4011/compiler/rustc_resolve/src/late.rs#L4797

https://github.com/rust-lang/rust/blob/717f5df2c308dfb4b7b8e6c002c11fe8269c4011/compiler/rustc_resolve/src/late.rs#L4875